integration/test/MLS/Util.hs

```diff
@@ -761,6 +759,14 @@ createApplicationMessage cid messageContent = do
 setMLSCiphersuite :: Ciphersuite -> App ()
 setMLSCiphersuite suite = modifyMLSState $ \mls -> mls {ciphersuite = suite}
 
+withCiphersuite :: HasCallStack => Ciphersuite -> App a -> App a
+withCiphersuite suite action = do
+  suite0 <- (.ciphersuite) <$> getMLSState
+  setMLSCiphersuite suite
+  r <- action
+  setMLSCiphersuite suite0
```

Comment:
Yeah, normally one would have to use bracket here to be exception-safe (also wrt async exceptions). I wrote this sloppily because we don't really recover from exceptions within tests (beyond printing diagnostics). But you're probably right, it doesn't really cost us much to make this safe.